### PR TITLE
add range check support for snarkyrs

### PR DESCRIPTION
**Pull request by fabrizio-m**
Adding support to use range checks in snarkyrs

### CODE REVIEW

**Comment by dannywillems** [comment]
@fabrizio-m can you make the CI happy please? :)

File: kimchi/src/snarky/range_checks.rs [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+            let v: FieldVar<Fp> = sys.compute(loc!(), |_| *private.unwrap())?;
+
+            sys.range_check(loc!(), v.clone(), v.clone(), v)?;
+
+            Ok(())
+        }
+    }
+
+    #[test]
+    fn test_bits() {
```
Comment (dannywillems) [inline kimchi/src/snarky/range_checks.rs:203]: Is it a test?

Comment (dannywillems) [reply kimchi/src/snarky/range_checks.rs:203]: I would remove it as it is not a test and it is not related to range_checks.

File: kimchi/src/snarky/constraint_system.rs [hunk 2]
```diff
@@ -1649,6 +1651,38 @@ impl<Field: PrimeField> SnarkyConstraintSystem<Field> {
                     self.add_row(labels, loc, vars, GateType::EndoMulScalar, vec![]);
                 }
             }
+            KimchiConstraint::RangeCheck(rows) => {
+                let rows: Result<[Vec<Cvar>; 4], _> = rows.try_into();
+                let rows: Result<[[Cvar; 15]; 4], _> = rows.map(|rows| {
```
Comment (dannywillems) [inline kimchi/src/snarky/constraint_system.rs:1656]: Suggestion: I would use the constant from Kimchi for the number of columns.

**Review by volhovm — APPROVED** [review]

**Pull request merged**